### ISSUE TITLE
Should we have the changelog as part of the Sphinx documentation?

### ISSUE THREAD

**Issue by DougBurke** [sherpa/sherpa#555]
AstroPy (and others) have a changelog in the documentation, which I find very useful (we do this for CIAO after all ;-).

There's a summary page

http://docs.astropy.org/en/stable/whatsnew/3.0.html

and a detailed page

http://docs.astropy.org/en/stable/changelog.html#changelog

Can we do this for the Sherpa Sphinx documentation?

**Comment by hamogu** [comment]
+1 - I find it useful. It does require discipline (or a bot to check) to make sure every PR adds to the changelog and in astropy it requires a lot of rebases, because often PRs add to the same section of the changelog.

**Comment by hamogu** [comment]
Update: Astropy changed to towncrier for their changelog system: https://pypi.org/project/towncrier/
(or see the astropy developer docs: https://docs.astropy.org/en/latest/development/workflow/maintainer_workflow.html#updating-and-maintaining-the-changelog). That add one step to the release procedure (run towncrier to turn the news fragments files into the sorted readme), but no rebases are ever needed now for changes to the changelog.

**Comment by hamogu** [comment]
@DougBurke @Marie-Terrell : Food for thought post release.

Adding some more details since this issue is so old, that the original reference to astropy does not refer to the same CHANGELOG system that is used to refer to. 

I have used two different systems for changelogs in the past, that write the main changelog as part of each PR and allow for automatic corss-linking from the changelog to the relevant GH PR, so that it's easy to go back and read the details discussion, see the code diff on the GH web interface etc.
Both approaches have been used by astropy at some point. I'm going to summarize pros and cons now. There certainly are other ways to deal with that.

1. **Each PR adds a few lines to a file CHANGES.rst in the repro and this file is included in the sphinx build**. An example for such a PR is here: https://github.com/Chandra-MARX/marxs/pull/221/files  The CHANGES.rst file is then included in the doc build, so that you can see the rendered version here: https://marxs.readthedocs.io/en/latest/changelog.html with a link to the original PR. It works so well and it so stable that I've already forgotton which sphinx extension is actually responsible for resolving the link, I suspect that it's the "edit_on_github" extension (https://github.com/Chandra-MARX/marxs/blob/main/docs/conf.py#L181), but we could find out. The beauty of this approach is the simplicity: The Changelog is part of the PR, so it gets merged at the same time and it gets reviewed at the same time, presumably by people who know why this is important. As soon as a PR is merged and the docs are rebuild, the CHANGELOG is there, it's linked. If something needs to be changed (typo in changelog etc, PR included in master, but not in a release) it's just another PR, no extra workflow, no external document to keep up to date. Just before a release, someone needs to update the title of the relevant section (e.g. from "1.3 unreleased" to "1.3 released on 12-12-2012") and then just after a release, someone has to make a new section ("1.4 unreleased"). The downside is that each PR edits the same file (CHANGES.rst). If there are only few sections, many PRs will add to the same location (e.g. to "bug fixes") causing the need to frequent rebasing. Those rebases are easy (just move the lines on top of whatever has been merged in the meantime), but annoying.
2. **towncrier**: See references in the comment above. Here, agian the CHANGELOG is put in as part of the PR, but each in a separate file and those files are then sorted and merged into a single changelog file by a separate package (towncrier) just before a release. This means some extra complexity (run the towncrier script once per release), but, since each PR add a new file, no risk of merge conflicts.

For small projects 1 works fine, for astropy the rebases were bad enough that we switched to 2, which has been used successfully for a few releaes now. I suspect that sherpa is somewhere in the middle.

**Comment by DougBurke** [comment]
@hamogu - in system 1 I can see that it's "easy" to add extra information to the changelog - e.g. to highlight certain changes - as it's just manually editing the file. I assume we can do this in system 2 as we use `towncrier` to generate the text and then would manually edit the file afterwards (in a separate PR probably) to add the extra information. Or is my mental model all wrong here?

**Comment by hamogu** [comment]
You are correct - you could hand-edit the towncrier generated file after it's been generated.
